c10/macros/Macros.h

```diff
@@ -118,7 +118,7 @@
 #define C10_HAS_CPP_ATTRIBUTE(x) (0)
 #endif
 
-#ifndef FBCODE_CAFFE2
+#if !defined(FBCODE_CAFFE2) && !defined(C10_NO_DEPRECATED)
```

Comment:
Nit (C10_NO_DEPRECATED is not define in fbcode, is it?)
```suggestion
#if !defined(C10_NO_DEPRECATED)
```